lib/streamlit/hashing.py

```diff
@@ -156,7 +159,19 @@ def _hashing_error_message(start):
         To stop this warning from showing in the meantime, try one of the following:
         * **Preferred:** modify your code to avoid using this type of object.
         * Or add the argument `ignore_hash=True` to the `st.cache` decorator.
-    """ % {'start': start}).strip("\n")
+    """
+        % {"start": start}
+    ).strip("\n")
+
+
+def _get_main_script_directory():
+    """Get the directory of the main script.
+    """
+    import __main__
+    import os
+
+    main_path = __main__.__file__
+    return os.path.dirname(main_path)
```

Comment:
I am assuming this happens while I am executing a streamlit script and this is well defined.